simple_parsing/parsing.py

```diff
@@ -49,20 +51,33 @@ def __init__(
             "--no-cache" and "--no_cache" can both be used to point to the same
             attribute `no_cache` on some dataclass.
 
-        - add_dest_to_option_strings: bool, optional
+        - argument_generation_mode : ArgumentGenerationMode, optional
 
-            Whether or not to add the `dest` of each field to the list of option
-            strings for the argument.
-            When True (default), each field can be referenced using either the
-            auto-generated option string or the full 'destination' of the field
-            in the resulting namespace.
-            When False, only uses the auto-generated option strings.
+            How to generate the arguments. In the ArgumentGenerationMode.FLAT mode,
+            the default one, the arguments are flat when possible, ignoring
+            their nested structure and including it only on the presence of a
+            conflict.
 
-            The auto-generated option strings are usually just the field names,
-            except when there are multiple arguments with the same name. In this
-            case, the conflicts are resolved as determined by the value of
-            `conflict_resolution` and each field ends up with a unique option
-            string.
+            In the ArgumentGenerationMode.NESTED mode, the arguments are always
+            composed reflecting their nested structure.
+
+            In the ArgumentGenerationMode.BOTH mode, both kind of arguments
+            are generated.
+
+        - nested_mode : NestedMode, optional
+
+            How to handle argument generation in for nested arguments
+            in the modes ArgumentGenerationMode.NESTED and ArgumentGenerationMode.BOTH.
+            In the NestedMode.DEFAULT mode, the nested arguments are generated
+            reflecting their full 'destination' path from the returning namespace.
+
+            In the NestedMode.WITHOUT_ROOT, the first level is removed. This is useful
+            because sometimes the first level is uninformative. For example,
+            'args'  is redundant and worthless for the arguments
```

Comment:
It would only be redundant and uninformative if you added arguments for a single all-encompassing dataclass. Maybe just reword this a little bit? Since from reading this, new users could think that `args` will always be the root.

Reply:
Something like "This is useful when `parser.add_arguments` is only called once,  and where the same prefix would be shared by all arguments."

Reply:
@lebrice it makes sense. Addressed here: https://github.com/lebrice/SimpleParsing/pull/117/commits/a7f72239d8a2ec0f7d17caff8225a320cd1de6de